model/vsc_parser.py

```diff
@@ -289,7 +308,9 @@ def ast_Arguments(parent, yamltree, argtype = 'in_arguments') -> list[Argument]:
     for st in subtrees:
         node = Argument(get_yaml_value(st, 'name'), parent)
         node.description = get_yaml_value(st, 'description')
-        node.type = get_yaml_value(st, 'datatype')
+        node.datatype = get_yaml_value(st, 'datatype')
```

Comment:
We should preferably check how this change affect existing templates. For example, we have a regression for https://github.com/COVESA/vsc-tools/blob/master/templates/simple_overview.tpl  as it refers to `{{x.type}}`. Have not checked output of other templates if there are any more regressions.

(With this PR we can later extend templates to also consider error information, but for now it is enough that we do not introduce any regressions)

Reply:
> We should preferably check how this change affect existing templates

Yes.  The "existing templates" in /templates are controlled by the same code project and the intention was to update them accordingly, of course.  In other words, it's a bug that the change to the template was not included in the push.

Reply:
Daniel Wilms and have done some prototyping for a test framework for VSS to detect regressions (see e.g. https://github.com/COVESA/vss-tools/pull/159, Daniel has been working on a similar concept using Python). Would be great if we later could add something similar so VSC-tools so regressions to templates would be detected by CI.

Reply:
Agree, that could be useful, probably in combination with traditional unit tests based on input + expected output!
I would probably build out some more traditional tests first and add this "diffing against previous output" when there are more feature-full templates, and things have stabilized a bit more.